### PR TITLE
Entra ID integration: integration script updates and web onboarding prerequisites

### PR DESCRIPTION
**Pull request by justinas**
This (together with the Enterprise counterpart https://github.com/gravitational/teleport.e/pull/4273) is the last large changeset for the initial version of Entra ID integration.

This PR:
* Updates Prehog protos with values for Entra ID (as introduced in https://github.com/gravitational/cloud/pull/9111)
* Fixes a couple of bugs in the integration script discovered while testing end-to-end.
  * Onboarding script now calls `az login` to make sure we have the necessary token to make calls to the "private" Azure API during onboarding. 
* Adds prerequisites for Enterprise web changes

changelog: Added support for Microsoft Entra ID directory synchronization (Teleport Enterprise only, preview)

### CODE REVIEW

File: api/proto/teleport/usageevents/v1/usageevents.proto [hunk 1]
```diff
@@ -562,7 +563,12 @@ enum IntegrationEnrollKind {
   INTEGRATION_ENROLL_KIND_MACHINE_ID_GITLAB = 15;
   INTEGRATION_ENROLL_KIND_MACHINE_ID_JENKINS = 16;
   INTEGRATION_ENROLL_KIND_MACHINE_ID_ANSIBLE = 17;
-  INTEGRATION_ENROLL_KIND_SERVICENOW = 18;
+  INTEGRATION_ENROLL_KIND_MACHINE_ID_AWS = 18;
```
Comment (justinas) [inline api/proto/teleport/usageevents/v1/usageevents.proto:566]: `18` is `MACHINE_ID_AWS` in prehog (both cloud and teleport), but `SERVICENOW` here. I guess this is a mistake due to us manually synchronizing `usageevents.proto` to prehog as a source of truth? Not sure how to proceed.

File: web/packages/teleport/src/services/sales/index.ts [hunk 2]
```diff
@@ -26,6 +26,8 @@ const UPGRADE_TEAM_URL = 'https://goteleport.com/r/upgrade-team';
 const UPGRADE_COMMUNITY_URL = 'https://goteleport.com/r/upgrade-community';
 // UPGRADE_IGS_URL is enterprise upgrading to enterprise with Identity Governance & Security
 const UPGRADE_IGS_URL = 'https://goteleport.com/r/upgrade-igs';
+// UPGRADE_POLICY_URL is enterprise upgrading to enterprise with Policy
+export const UPGRADE_POLICY_URL = 'https://goteleport.com/r/upgrade-policy';
```
Comment (justinas) [inline web/packages/teleport/src/services/sales/index.ts:30]: Added in https://github.com/gravitational/next/pull/2505

File: e
Comment (jakule) [inline e:1]: Is this intentional? From what I see this points to your dev branch and not `master`

File: web/packages/teleport/src/services/sales/index.ts [hunk 3]
```diff
@@ -40,12 +42,15 @@ function getParams(
...
+  event?: CtaEvent,
+  url?: string
 ) {
-  let url = UPGRADE_COMMUNITY_URL;
-  if (isEnterprise) {
-    // TODO(mcbattirola): remove isTeam when it is no longer used
-    url = cfg.isTeam ? UPGRADE_TEAM_URL : UPGRADE_IGS_URL;
+  if (!url) {
+    url = UPGRADE_COMMUNITY_URL;
+    if (isEnterprise) {
```
Comment (jakule) [inline web/packages/teleport/src/services/sales/index.ts:50]: nit:
```suggestion
  if (!url && isEnterprise) {
    url = UPGRADE_COMMUNITY_URL;
```

File: e
Comment (justinas) [reply e:1]: Not intentional, will revert 👍

**Review by kimlisa — APPROVED** [review]

File: web/packages/teleport/src/services/sales/index.ts (hunk 2 above)
Comment (justinas) [reply web/packages/teleport/src/services/sales/index.ts:30]: We only need to change the redirect destination in `next`, the URLs in the frontend can stay, right?

**Review by jakule — APPROVED** [review]

**Pull request merged**